### DISCUSSION TITLE
std::string to JSON parsing

### DISCUSSION THREAD

**Discussion by cxxboy** [nlohmann/json#2450] [Q&A] [answered]
I have a request coming in as JSON, Example:
`curl --header "Content-Type: application/json" -d '{"hello":123}' http://127.0.0.1:8766/poster`

The std::string contains the `{"Hello":123}` I can able to print out. 
But, How can I parse that as JSON object?

Here is something I tried out:

```c++
using json = nlohmann::json;
json::parser_callback_t json_cb = [](int depth, json::parse_event_t event, json& parsed){
	if(event == json::parse_event_t::key and parsed == json("hello")){
		return true;
	}else{
		return false;
	}
};
// Here, INPUT_TEXT is std::string type (which is converted from char* to std::string from raw socket read() )
auto parsed_json = json::parse(INPUT_TEXT, json_cb);
std::cout << parsed_json << std::endl;
```
For the note, `parsed_json["hello"];` also gives me null (does not work)

But this keeps giving me NULL. Can someone help me figure this out? Or is there any other way I can parse this other than callback (I dont fully understand the use of callbacks in the json::parser, Im copy-pasting from docs?

**Comment by nlohmann** [comment, marked as answer]
You can just call `auto parse_json = json::parse(INPUT_TEXT);`. The callback is optional, and you may only need it if you want to filter values while parsing.

**Reply by nlohmann** [reply]
Did this solve your issue?
